### DISCUSSION TITLE
I've published a userscript to bring the old feed back

### DISCUSSION THREAD

**Discussion by Sec-ant** [community/community#66260] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Body

Source code repo: https://github.com/Sec-ant/github-feed-back

Greasy fork link: https://greasyfork.org/zh-CN/scripts/474720-github-feed-back

For people who just want the code:

```js
// ==UserScript==
// @name         GitHub Feed Back
// @namespace    https://github.com/Sec-ant
// @version      0.1.5
// @author       Ze-Zheng Wu
// @description  Bring my GitHub feed back
// @license      MIT
// @icon         https://github.githubassets.com/favicons/favicon.svg
// @homepage     https://github.com/Sec-ant/github-feed-back
// @homepageURL  https://github.com/Sec-ant/github-feed-back
// @source       https://github.com/Sec-ant/github-feed-back.git
// @supportURL   https://github.com/Sec-ant/github-feed-back/issues
// @downloadURL  https://registry.npmmirror.com/github-feed-back/latest/files/dist/github-feed-back.user.js
// @updateURL    https://registry.npmmirror.com/github-feed-back/latest/files/dist/github-feed-back.meta.js
// @match        https://github.com/
// @match        https://github.com/?*
// @match        https://github.com/dashboard*
// @run-at       document-start
// ==/UserScript==

(function () {
  'use strict';

  let isIncludeFragmentFound = false;
  let isFilterButtonFound = false;
  const observer = new MutationObserver((_, observer2) => {
    const includeFragment = document.querySelector(
      '[data-target="feed-container.content"] > include-fragment'
    );
    if (includeFragment) {
      includeFragment.setAttribute("src", "/dashboard-feed");
      isIncludeFragmentFound = true;
    }
    const filterButton = document.querySelector(
      '[data-target="feed-container.feedTitle"] + div'
    );
    if (filterButton) {
      filterButton.remove();
      isFilterButtonFound = true;
    }
    if (isIncludeFragmentFound && isFilterButtonFound) {
      observer2.disconnect();
    }
  });
  observer.observe(document.documentElement, {
    childList: true,
    subtree: true
  });
  document.addEventListener("DOMContentLoaded", () => {
    observer.disconnect();
  });

})();
```

Screenshot:

![image](https://github.com/community/community/assets/10386119/4b13a4da-9088-4b6b-ac38-9248f850d7f7)

**Comment by Gerrit0** [comment]
Hey, I did this too last time GitHub did this to us! <https://github.com/Gerrit0/old-github-feed> - one thing your implementation misses is that it breaks if the user tries to hit the "next page" button at the bottom of the feed

**Reply by Sec-ant** [reply]
Hey, I can't find a "next page" button but a "More" button, and I didn't notice any breaks when I hit the "More" button.

**Reply by Gerrit0** [reply]
Shoot, my bad! I saw that you didn't disable the mutation observer when the user clicked on it, which broke things initially for me, seems like a cleaner implementation if targeting `include-fragment` lets you avoid that

**Reply by Sec-ant** [reply]
Yeah it works out smoothly, guess that beginner's luck :)

**Comment by sweetbbak** [comment]
thanks! The new feed is horrendous. I just want to see when the people I follow create repo's and star repo's.

**Comment by falsandtru** [comment]
It is useless. This old feed will be deleted. The only effectual way is to revert the official feed.

**Reply by Sec-ant** [reply]
Yep, but I'm not competent to do that. In the meantime I'll keep this maintained until GitHub kills the old feed or starts to listen to the users.

**Comment by zzy-life** [comment]
Good job, at least I can use the old version now

**Comment by dmarx** [comment]
doing the lord's work

**Comment by Norod** [comment]
Thank you very much for this!

**Comment by pouriap** [comment]
Thank you for this. I hope user feedback will be able to overcome corporate greed and we'll get the official old feed back, otherwise this will probably stop working soon.

**Comment by weearc** [comment]
Thank you, thank you for this. You saved us a lot of time. Thank you.

**Comment by weijiang1994** [comment]
Very nice!!! Thank you. :smile:

**Comment by JPeisach** [comment]
W. I thought extensions like this didn't exist for Safari. W man, looking out for everyone.

**Reply by SamJakob** [reply]
Ironically my Safari extension for this was the only one I've managed to fully release so far lol

**Comment by tborychowski** [comment]
this doesn't seem to work for me (or they disabled the old feed?)
I can still see the new BS, just the filter buttons are hidden.
I suppose now, the best thing to do, is to open filters, uncheck all and save... ;-)
